src/dataflow/src/source/postgres.rs

```diff
@@ -197,26 +197,47 @@ impl PostgresSourceReader {
             .parse()
             .or_else(|_| Err(anyhow!("invalid lsn"))));
         for info in publication_tables {
-            // TODO(petrosagg): use a COPY statement here for more efficient network transfer
-            let data = client
-                .simple_query(&format!(
-                    "SELECT * FROM {:?}.{:?}",
-                    info.namespace, info.name
-                ))
+            let relation_id: Datum = (info.rel_id as i32).into();
+            let reader = client
+                .copy_out_simple(
+                    format!(
+                        "COPY {:?}.{:?} TO STDOUT (FORMAT TEXT, DELIMITER '\t')",
+                        info.namespace, info.name
+                    )
+                    .as_str(),
+                )
                 .await?;
-            for msg in data {
-                if let SimpleQueryMessage::Row(row) = msg {
+
+            pin_mut!(reader);
+            while let Some(Ok(b)) = reader.next().await {
+                // Convert raw rows from COPY into repr:Row. Each Row is a relation_id
+                // and list of string-encoded values, e.g. Row{ 16391 , ["1", "2"] }
+                let mut parser = pgwire::CopyTextFormatParser::new(b.as_ref(), "\t", "\\N");
```

Comment:
not sure if it's worth fiddling with, but we could potentially make this mutable so we don't need to create a new parser per row